src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseCreator.cs

```diff
@@ -42,11 +48,55 @@ public Task<bool> EnsureDeletedAsync(CancellationToken cancellationToken = defau
 
     /// <inheritdoc/>
     public bool EnsureCreated()
-        => clientWrapper.CreateDatabase(currentDbContext.Context.GetService<IDesignTimeModel>());
+    {
+        var created = clientWrapper.CreateDatabase(currentDbContext.Context.GetService<IDesignTimeModel>());
+        if (created)
+        {
+            SeedFromModel();
+        }
+        return created;
+    }
 
     /// <inheritdoc/>
-    public Task<bool> EnsureCreatedAsync(CancellationToken cancellationToken = default)
-        => clientWrapper.CreateDatabaseAsync(currentDbContext.Context.GetService<IDesignTimeModel>(), cancellationToken);
+    public async Task<bool> EnsureCreatedAsync(CancellationToken cancellationToken = default)
+    {
+        var created = await clientWrapper.CreateDatabaseAsync(currentDbContext.Context.GetService<IDesignTimeModel>(), cancellationToken).ConfigureAwait(false);
+        if (created)
+        {
+            await SeedFromModelAsync(cancellationToken).ConfigureAwait(false);
+        }
+        return created;
+    }
+
+    /// <summary>
+    /// Inserts any data that was added to the model using an overload of <see cref="EntityTypeBuilder{TEntity}.HasData(TEntity[])"/>
+    /// </summary>
+    public void SeedFromModel()
+        => database.SaveChanges(AddModelData().GetEntriesToSave());
+
+    /// <summary>
+    /// Inserts any data that was added to the model using an overload of <see cref="EntityTypeBuilder{TEntity}.HasData(TEntity[])"/>
+    /// </summary>
+    /// <param name="cancellationToken">A cancellation token</param>
+    /// <returns>A task representing this async operation.</returns>
+    public async Task SeedFromModelAsync(CancellationToken cancellationToken = default)
+        => await database.SaveChangesAsync(AddModelData().GetEntriesToSave(), cancellationToken).ConfigureAwait(false);
+
+    private IUpdateAdapter AddModelData()
+    {
+        var updateAdapter = updateAdapterFactory.CreateStandalone();
+        foreach (var entityType in designTimeModel.Model.GetEntityTypes())
```

Comment:
Do we not need to filter this on root entities only so it doesn't attempt to seed owned ones?

Reply:
Good question--let me try it. It may be that owned entities never return anything from GetSeedData.